utils/data-loading.ts

```diff
@@ -31,12 +31,16 @@ export function useDataSource<Data, Deps extends DependencyList | undefined, E e
     // For more information see: https://juliangaramendy.dev/use-promise-subscription/
     let isSubscribed = true
     // For isSubscribed to work with reloading we need to use reload as a Ref
-    reloadRef.current = () => {
+    reloadRef.current = async () => {
       setIsLoading(true)
-      createDataPromise()
-        .then((data) => isSubscribed && setData(data))
-        .catch((error) => isSubscribed && setError(error))
-        .finally(() => isSubscribed && setIsLoading(false))
+      try {
+        const data = await createDataPromise()
+        if (isSubscribed) setData(data)
+      } catch (error) {
+        if (isSubscribed) setError(error)
+      } finally {
+        if (isSubscribed) setIsLoading(false)
+      }
```

Comment:
I rewrote this as a try/catch so coverage requirements would catch that the error case wasn't being tested. Previously, it considered it all one line.

Reply:
Maybe we can also make the `isSubscribed` case more explicit?

```suggestion
      if (isSubscribed) {
        try {
          setData(await createDataPromise())
        } catch (error) {
          setError(error)
        } finally {
          setIsLoading(false)
        }
      } else {
        try {
          await createDataPromise()
        } catch (error) {}
      }
```

Reply:
@yanirs I don't think this will work as it is possible for `isSubscribed` to have changed while waiting for the promise resolution, and we want the value at the execution time of `setX`. Unfortunately the new await syntax makes it a little less obvious that this is happening in a different "queued sub-task", and where exactly that line is.

@withinboredom I am surprised the inline `if` statements here pass our code conventions? We don't do that anywhere AFAIK.

Reply:
> I am surprised the inline if statements here pass our code conventions? We don't do that anywhere AFAIK.

I always write inline ifs unless it's more than one statement and let autoformat / linters fix them if they aren't allowed (habit from writing C/C++ for years) but maybe the rules changed with the updated eslint and it wasn't caught. I'll update the code here.

Reply:
Yeah I generally prefer having stable syntax (syntax that doesn't change if the semantics doesn't), this means less editing and better diffs.